Applications/Utils/MeshEdit/ResetPropertiesInPolygonalRegion.cpp

```diff
@@ -0,0 +1,215 @@
+	for (std::size_t j(0); j < nodes.size(); j++)
+		rotated_nodes.push_back(new GeoLib::Point(nodes[j]->getCoords()));
+	// 2 rotate the Points
+	MathLib::DenseMatrix<double> rot_mat(3,3);
+	GeoLib::computeRotationMatrixToXY(normal, rot_mat);
+	GeoLib::rotatePoints(rot_mat, rotated_nodes);
+	// 3 set z coord to zero
+	std::for_each(rotated_nodes.begin(), rotated_nodes.end(),
+		[] (GeoLib::Point* p) { (*p)[2] = 0.0; }
+	);
+
+	// *** mark rotated nodes
+	std::vector<bool> outside(rotated_nodes.size(), true);
+	for (std::size_t k(0); k<rotated_nodes.size(); k++) {
+		if (rot_polygon.isPntInPolygon(*(rotated_nodes[k]))) {
+			outside[k] = false;
+		}
+	}
+
+	for (std::size_t j(0); j < rotated_nodes.size(); j++)
+		delete rotated_nodes[j];
+
+	std::vector<GeoLib::Point*> & rot_polygon_pnts(
+		const_cast<std::vector<GeoLib::Point*> &>(
+			rot_polygon.getPointsVec()
+		)
+	);
+	for (std::size_t k(0); k < rot_polygon_pnts.size(); k++)
+		delete rot_polygon_pnts[k];
+
+	return outside;
+}
+
+void resetProperty(MeshLib::Mesh &mesh, GeoLib::Polygon const& polygon,
+	std::size_t new_property)
+{
+	std::vector<bool> outside(markNodesOutSideOfPolygon(mesh.getNodes(),
+		polygon));
+
+	for(MeshLib::Element * elem : mesh.getElements()) {
+		bool elem_out(true);
+		for(std::size_t k(0); k<elem->getNNodes() && elem_out; ++k) {
+			if (! outside[elem->getNode(k)->getID()]) {
+				elem_out = false;
+			}
+		}
+		if (elem_out) {
+			elem->setValue(new_property);
+		}
+	}
+}
+
+int main (int argc, char* argv[])
+{
+	LOGOG_INITIALIZE();
+	logog::Cout* logog_cout (new logog::Cout);
+	BaseLib::LogogSimpleFormatter *custom_format (new BaseLib::LogogSimpleFormatter);
+	logog_cout->SetFormatter(*custom_format);
+
+	TCLAP::CmdLine cmd("desc", ' ', "0.1");
+	TCLAP::ValueArg<std::string> mesh_in("i", "mesh-input-file",
+		"the name of the file containing the input mesh", true,
+		"", "file name");
+	cmd.add(mesh_in);
+	TCLAP::ValueArg<std::string> mesh_out("o", "mesh-output-file",
+		"the name of the file the mesh will be written to", true,
+		"", "file name");
+	cmd.add(mesh_out);
+	TCLAP::ValueArg<std::string> geometry_fname("g", "geometry",
+		"the name of the file containing the input geometry", true,
+		"", "file name");
+	cmd.add(geometry_fname);
+	TCLAP::ValueArg<unsigned> polygon_id_arg("p", "polygon-id",
+		"id of polygon in the geometry", true, 0, "number");
+	cmd.add(polygon_id_arg);
+	TCLAP::ValueArg<unsigned> new_material_id_arg("n", "new-material-id",
+		"new material id", false, 0, "number");
+	cmd.add(new_material_id_arg);
+	cmd.parse(argc, argv);
+
+	// *** read mesh
+	MeshLib::Mesh * mesh(FileIO::readMeshFromFile(mesh_in.getValue()));
+
+	// *** read geometry
+	GeoLib::GEOObjects geometries;
+	{
+		FileIO::BoostXmlGmlInterface xml_io(geometries);
+		if (xml_io.readFile(geometry_fname.getValue())) {
+			INFO("Read geometry from file \"%s\".",
+				geometry_fname.getValue().c_str());
+		} else {
+			ERR("Problems to read geometry from file \"%s\".",
```

Comment:
i think this msg is not necessary, the reader will output a message that this is not a valid GML file.